modules/openapi-generator/src/main/resources/typescript-fetch/modelOneOf.mustache

```diff
@@ -1,13 +1,12 @@
 {{#hasImports}}
+{{#imports}}
 import {
-    {{#imports}}
     {{{.}}},
-    {{/imports}}
-    {{#oneOf}}
```

Comment:
Why not keep the oneOf condition?

Reply:
It was introduced in https://github.com/OpenAPITools/openapi-generator/pull/4387

cc @eriktim

Reply:
Good question, I can probably add it back, I just copied the changes from the referenced issue; and the samples seem to not even use this template so I need to do some more testing.

Reply:
I've just tried with the oneOf and its causing some import errors during compile time

Reply:
@eriktim can you help here?

Reply:
>I've just tried with the oneOf and its causing some import errors during compile time

what are the errors?

Reply:
Can you also provide the spec file you used for testing?

Reply:
So this is the diff I'm testing with

```diff
diff --git a/modules/openapi-generator/src/main/resources/typescript-fetch/modelOneOf.mustache b/modules/openapi-generator/src/main/resources/typescript-fetch/modelOneOf.mustache
index 99ab3085ac3..8392d8f71e5 100644
--- a/modules/openapi-generator/src/main/resources/typescript-fetch/modelOneOf.mustache
+++ b/modules/openapi-generator/src/main/resources/typescript-fetch/modelOneOf.mustache
@@ -2,9 +2,10 @@
 {{#imports}}
 import {
     {{{.}}},
-    {{.}}FromJSON,
+    {{#oneOf}}
     {{.}}FromJSONTyped,
     {{.}}ToJSON,
+    {{/oneOf}}
 } from './{{.}}';
 {{/imports}}
 ```

With the schema from here https://github.com/goauthentik/authentik/blob/master/schema.yml

And using these options
```
--additional-properties=typescriptThreePlus=true,supportsES6=true,npmName=@goauthentik/api,npmVersion=1.2.3
```

With the `oneOf` added in it generates code like

```typescript
import {
    PlexAuthenticationChallenge,
    PlexAuthenticationChallengeFromJSONTyped,
    PlexAuthenticationChallengeToJSON,
    RedirectChallengeFromJSONTyped,
    RedirectChallengeToJSON,
} from './PlexAuthenticationChallenge';
import {
    RedirectChallenge,
    PlexAuthenticationChallengeFromJSONTyped,
    PlexAuthenticationChallengeToJSON,
    RedirectChallengeFromJSONTyped,
    RedirectChallengeToJSON,
} from './RedirectChallenge';
```
Which looks like its just looking at the wrong scope somewhere, i'll continue debugging this too.

Reply:
Thanks @BeryJu. It's hard to validate your spec file as it is huge. It's always advisable to reduce it to a [SSCCE](http://www.sscce.org/). For example, I am now somehow unable to debug using `--global-property debugModels`.
Can you try creating a new block for `oneOf` as well? Like this:
```
{{#oneOf}}
import {
    {{{.}}}FromJSON,
    {{{.}}}FromJSONTyped,
    {{{.}}}ToJSON,
} from './{{.}}';
{{/oneOf}}
```

Reply:
I've trimmed the schema down to a somewhat minimal test case here: https://gist.github.com/BeryJu/f0c2ac4486a59dbf29c19818a2cb2474

Also with the suggest block from you, i.e. a header looking like this
```
{{#hasImports}}
{{#oneOf}}
import {
    {{{.}}}FromJSON,
    {{{.}}}FromJSONTyped,
    {{{.}}}ToJSON,
} from './{{.}}';
{{/oneOf}}

{{/hasImports}}
{{>modelOneOfInterfaces}}
```

I get a good result of

```
import {
    PlexAuthenticationChallengeFromJSON,
    PlexAuthenticationChallengeFromJSONTyped,
    PlexAuthenticationChallengeToJSON,
} from './PlexAuthenticationChallenge';
import {
    RedirectChallengeFromJSON,
    RedirectChallengeFromJSONTyped,
    RedirectChallengeToJSON,
} from './RedirectChallenge';
```

instead of multiple imports from the same file